framework/sessions/__init__.py

```diff
@@ -44,6 +47,26 @@ def add_key_to_url(url, scheme, key):
     parsed_redirect_url = parsed_url._replace(**replacements)
     return urlparse.urlunparse(parsed_redirect_url)
 
+def get_node_id(url):
+    try:
+        node_id = [x for x in url.split("/") if x][2]
+        # Avoid circular import
+        from website.models import Node
+        if Node.find(Q('_id', 'eq', node_id)).count() == 1:
```

Comment:
If it can't be avoided, you should probably be iterating over `request.path.split('/')`, possibly checking length to see if it could be a guid, and returning the first successful `Node.load`'s `id`